Add GFS ICs to archive that are needed for GEFS

EricSinsky-NOAA opened the pull request:
# Description

This PR adds GFS IC files that GEFS needs to the GFS archive. The following is added to the GFS archive:

**enkfgfs_grp$N.tar (task - enkfgfs_earc_tars_$N)**
enkfgfs.${pdy}/${cyc}/${ENSMEM}/analysis/atmos/enkfgfs.t${cyc}z.ratmi003.nc
enkfgfs.${pdy}/${cyc}/${ENSMEM}/analysis/atmos/enkfgfs.t${cyc}z.ratmi009.nc

**gdas.tar (task - gdas_arch_tars)**
gdas.${pdyp}/${cycp6}/model/atmos/history/gdas.t${cycp6}z.atmf003.ensres.nc

**gdasocean_restart.tar (task - gdas_arch_tars)**
gdas.${pdyp}/${cycp6}/model/ocean/restart/${pdyp}.${cycp3}0000.MOM.res*

**enkfgdas_restartb_grp$N.tar (task - enkfgdas_earc_tars_$N)**
enkfgdas.${pdyp}/$cycp6/$ENSMEM/model/med/restart/${pdyp}.${cycp3}0000.ufs.cpld.cpl.r.nc

Resolves #3940 
Resolves #3908 
Resolves https://github.com/NOAA-EMC/GFS/issues/17

<!-- For more on writing good commit messages, see https://cbea.ms/git-commit/ -->

# Type of change
- [ ] Bug fix (fixes something broken)
- [x] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
<!-- Choose YES or NO from each of the following and delete the other -->
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO
  - [ ] EMC verif-global <!-- NOAA-EMC/EMC_verif-global#1234 -->
  - [ ] GDAS <!-- NOAA-EMC/GDASApp#1234 -->
  - [ ] GFS-utils <!-- NOAA-EMC/gfs-utils#1234 -->
  - [ ] GSI <!-- NOAA-EMC/GSI#1234 -->
  - [ ] GSI-monitor <!-- NOAA-EMC/GSI-Monitor#1234 -->
  - [ ] GSI-utils <!-- NOAA-EMC/GSI-Utils#1234 -->
  - [ ] UFS-utils <!-- ufs-community/UFS_UTILS#1234 -->
  - [ ] UFS-weather-model <!-- ufs-community/ufs-weather-model#1234 -->
  - [ ] wxflow <!-- NOAA-EMC/wxflow#1234 -->

# How has this been tested?
<!-- Please list any test you conducted, including the machine.

Example:
- Clone and build on WCOSS
- Cycled test on Orion
- Forecast-only on Hera
-->

- Tested the `C96C48mx500_S2SW_cyc_gfs` CI case on WCOSS2 and all tasks have succeeded with the archive changes. 

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have documented my code, including function, input, and output descriptions
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] This change is covered by an existing CI test or a new one has been added
- [ ] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [ ] I have made corresponding changes to the system documentation if necessary

EricSinsky-NOAA committed:
Add ratmi003 and ratmi009 to enkfgfs_grpN.tar

EricSinsky-NOAA committed:
Archive ocean restart files

JessicaMeixner-NOAA commented:
@EricSinsky-NOAA Thank you for helping with this!

EricSinsky-NOAA committed:
Add atmf${fhr3}.ensres.nc to archive

EricSinsky-NOAA committed:
Archive MED restart files for perturbed members

EricSinsky-NOAA committed:
Merge remote-tracking branch 'origin/develop' into issue3940

EricSinsky-NOAA committed:
Merge remote-tracking branch 'origin/develop' into issue3940

EricSinsky-NOAA commented:
Marking as ready for review. It should be noted that the gdas build in this PR fails on WCOSS2 due to pip being used in the jcb installation (issue #3933).

aerorahul reviewed (dismissed):
looks good to me.
@EricSinsky-NOAA thanks for this PR

EricSinsky-NOAA committed:
Merge remote-tracking branch 'origin/develop' into issue3940

DavidHuber-NOAA commented on parm/archive/master_gdas.yaml.j2:
```diff
@@ -39,7 +39,7 @@ datasets:
 {% if MODE == "cycled" %}
     # Determine if we will save restart ICs or not (only valid for cycled)
-    {% set save_warm_start_forecast, save_warm_start_cycled = ( False, False ) %}
+    {% set save_warm_start_forecast, save_warm_start_cycled = ( False, True ) %}
```
The warm start cycled restarts are quite large, hence why we usually only save them once every 4 days (16 cycles).  Does this really need to be saved for every cycle?  If so, perhaps we can work in a parameter to config.base or config.arch_tars that will tell the workflow to save the warm restarts every cycle if needed.

EricSinsky-NOAA replied:
@DavidHuber-NOAA GEFS only needs the gdas ocean restart data from `gdasocean_restart.yaml.j2` for `06z` and `18z` when GFS runs its retrospectives. If possible, it would be best if this can happen by default in the archive task. How about if I modify `parm/archive/master_gdas.yaml.j2` so that `gdasocean_restart.yaml.j2` will be used regardless of `save_warm_start_cycled`?

DavidHuber-NOAA replied:
@EricSinsky-NOAA That sounds great.  I have no problem with archiving the ocean restarts every cycle.

EricSinsky-NOAA replied:
@DavidHuber-NOAA Thank you for your suggestion. I have set `save_warm_start_cycled` back to `False` in  [3e246ec](https://github.com/NOAA-EMC/global-workflow/pull/3949/commits/3e246ec2a9ea4701bc14bfb7be9f12798d42fe55). I have also moved where `gdasocean_restart.yaml.j2` is included so that it does not depend on `save_warm_start_cycled`.

JessicaMeixner-NOAA replied:
@CatherineThomas-NOAA  was also thinking at one point to only save these for the subset of ensembles instead of all 80 ensembles to also save space.  Perhaps that can be added here as well?   I think there should be a formula of some sort to know which ensemble members need to be saved.

DavidHuber-NOAA replied:
@JessicaMeixner-NOAA This is for just the deterministic GDAS, not the ensemble.  Also, FYI, Eric is on leave through 8/28.

JessicaMeixner-NOAA replied:
Thanks for that clarification @DavidHuber-NOAA

EricSinsky-NOAA committed:
Set save_warm_start_cycled to False by default

The save_warm_start_cycled flag is set to false by default. The gdas ocean restart data is archived despite whether save_warm_start_cycled is true or false.

The gdas ocean restart data is archived regardless of the save_warm_start_cycled flag.

DavidHuber-NOAA reviewed (dismissed):
Looks good!  Thanks @EricSinsky-NOAA!

DavidHuber-NOAA commented:
Running CI on C6.

DavidHuber-NOAA commented:
@EricSinsky-NOAA are the ensemble ocean restarts also needed for GEFS?

bingfu-NOAA commented:
We have one more file missing. It is related to this issue. https://github.com/NOAA-EMC/global-workflow/issues/3895
Can we hold this PR until #3895 is resolved so that we can include this file in the archive list?

JessicaMeixner-NOAA commented:
@bingfu-NOAA would an alternative be to add to issue #3895 that this file also needs to be archived?

bingfu-NOAA commented:
> @bingfu-NOAA would an alternative be to add to issue #3895 that this file also needs to be archived?

I added a comment in #3895

EricSinsky-NOAA commented:
> @EricSinsky-NOAA are the ensemble ocean restarts also needed for GEFS?

@DavidHuber-NOAA Ensemble ocean restarts are also needed for GEFS, but in my tests, they are already being archived in `enkfgdas_restartb_grp${N}.tar` using the `master_enkf.yaml` by default.

DavidHuber-NOAA commented:
@EricSinsky-NOAA Indeed, I see that now:

https://github.com/NOAA-EMC/global-workflow/blob/d7e815218b4e117bda44519736808c717170eb19/parm/archive/enkf_restartb_grp.yaml.j2#L45-L51

Thanks!

EricSinsky-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into issue3940

EricSinsky-NOAA committed:
Archive ensres file only when DOHYBVAR is YES

EricSinsky-NOAA commented:
The two CI tests that failed (`C48mx500_3DVarAOWCDA` and `C96_atm3DVar`) have `DOHYBVAR=NO`, which seems to prevent `gdas.t18z.atmf003.ensres.nc `and `gdas.t18z.atmf006.ensres.nc` from being generated. I added a condition in `parm/archive/gdas.yaml.j2` ([1e8427e](https://github.com/NOAA-EMC/global-workflow/pull/3949/commits/1e8427e059049e4f78af8b74f5cbfc5611c6d44c)) so that these ensres files are only archived if `DOHYBVAR` is `YES`. This has resolved the CI failures in my offline testing.

DavidHuber-NOAA reviewed (dismissed):
Looks good!

DavidHuber-NOAA committed:
Merge branch 'develop' into issue3940

DavidHuber-NOAA commented:
@EricSinsky-NOAA did you run the full suite on C6?  If so, we can forgo running another set.

EricSinsky-NOAA commented:
> @EricSinsky-NOAA did you run the full suite on C6? If so, we can forgo running another set.

@DavidHuber-NOAA I did not run the full suite on C6. I only ran the problematic GFS cases on WCOSS2. When you get a chance, can you run the full set on C6?

DavidHuber-NOAA commented:
Sure, I'll go ahead and launch them.

EricSinsky-NOAA commented:
`C96C48mx500_S2SW_cyc_gfs` failed in the CI on C6 because `gdas.t18z.atmf003.ensres.nc` and `gdas.t18z.atmf006.ensres.nc` were generated about 30 seconds after the `gdas_arch_tar_gdas` task attempted to access these files. This issue appears intermittent because it did not occur in my offline testing. This suggests a possible dependency problem, which I will investigate further.

EricSinsky-NOAA committed:
Fixed ensres dependency issue

EricSinsky-NOAA committed:
Merge remote-tracking branch 'origin/issue3940' into issue3940

EricSinsky-NOAA commented:
After investigating, I found that `gdas.t18z.atmf003.ensres.nc` and `gdas.t18z.atmf006.ensres.nc` are generated in the `enkfgdas_echgres` task, which runs after `gdas_arch_tar_gdas`. Since `gdas_arch_tar_gdas` (using `gdas.yaml.j2`) was trying to archive them too early, I moved the files to `enkf.yaml.j2` (https://github.com/NOAA-EMC/global-workflow/pull/3949/commits/d3ed25a71508cc51893ad56ce668c4325563eb82), used by `enkfgdas_earc_tars_00`. This task runs after `enkfgdas_echgres`, avoiding dependency issues. The `C96C48mx500_S2SW_cyc_gfs` test passed on WCOSS2. Additional tests are running.

EricSinsky-NOAA commented:
All GFS CI tests have succeeded on WCOSS2.  Re-requesting review.

EricSinsky-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into issue3940

EricSinsky-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into issue3940

DavidHuber-NOAA reviewed (approved):
Looks good!

DavidHuber-NOAA commented:
Marking CI as passed on WCOSS2 after @EricSinsky-NOAA shared that all tests launched successfully on Cactus.  Launching CI on Hera.

DavidHuber-NOAA commented:
FYI @TerrenceMcGuinness-NOAA, I'm launching CI on Hera now.

DavidHuber-NOAA committed:
Merge branch 'develop' into issue3940

DavidHuber-NOAA commented:
All tests passed on Hera and WCOSS2.  Merging.
